cmd/kube-scheduler/app/testing/testserver.go

```diff
@@ -66,6 +67,7 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 		if len(result.TmpDir) != 0 {
 			os.RemoveAll(result.TmpDir)
 		}
+		configz.Delete("componentconfig")
```

Comment:
Where is this coming from?

Reply:
because we call `app.Setup` now, and it creates a configz, we need to clean up this config for next test case, or just log the error and don't return directly

```
	if cz, err := configz.New("componentconfig"); err == nil {
		cz.Set(cc.ComponentConfig)
	} else {
		return nil, nil, fmt.Errorf("unable to register configz: %s", err)
	}
```

Reply:
I think we need to reach to the OWNERS of the users of this library (`/pkg/integration/serving`) to learn what are their expectations. Perhaps we can change this to be much closer to `runCommand`. Otherwise, anything that is not common, should go outside of `Setup`.

Reply:
Actually, if we move this to `Run`, we still need to add `configz.Delete("componentconfig")` because `app.Run` is also called by `StartTestServer`.

We can find similar code in `cmd/kube-controller-manager/app/controllermanager.go` which also has a testserver used by `/pkg/integration/serving`

```
func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
	// To help debugging, immediately log version
	klog.Infof("Version: %+v", version.Get())

	if cfgz, err := configz.New(ConfigzName); err == nil {
		cfgz.Set(c.ComponentConfig)
	} else {
		klog.Errorf("unable to register configz: %v", err)
	}
```

it logs an error and continue the execution, otherwise it will also need a `configz.Delete`

Reply:
That is true. But at least we don't need to worry about it when testing Setup. If this is what other binaries are doing for the serving test, that's great.